onnxruntime_training_test.go

```diff
@@ -0,0 +1,328 @@
+package onnxruntime_go
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"path"
+	"testing"
+)
+
+func TestTrainingNotSupported(t *testing.T) {
+	InitializeRuntime(t)
+	defer CleanupRuntime(t)
+
+	options, e := NewSessionOptions()
+	if e != nil {
+		t.Logf("Failed creating options: %s\n", e)
+		t.FailNow()
+	}
+
+	trainingSession, e := NewTrainingSession("test_data/onnxruntime_training_test/training_artifacts/checkpoint",
+		"test_data/onnxruntime_training_test/training_artifacts/training_model.onnx",
+		"test_data/onnxruntime_training_test/training_artifacts/eval_model.onnx",
+		"test_data/onnxruntime_training_test/training_artifacts/optimizer_model.onnx",
+		nil, nil,
+		options)
+	if ok := errors.Is(e, trainingNotSupportedError); !ok {
+		t.Logf("Creating training session when onnxruntime lib does not support it should return error.")
+		t.FailNow()
+	}
+	defer func(session *TrainingSession) {
+		err := errors.Join(
+			trainingSession.Destroy(),
+		)
+		if err != nil {
+			t.Fail()
+		}
+	}(trainingSession)
+}
+
+func TestSessionInfo(t *testing.T) {
+	InitializeRuntime(t)
+	defer CleanupRuntime(t)
+
+	if !IsTrainingSupported() {
+		t.Skipf("Training is not supported on this platform/onnxruntime build.")
+	}
+
+	options, e := NewSessionOptions()
+	if e != nil {
+		t.Logf("Failed creating options: %s\n", e)
+		t.FailNow()
+	}
+	artifactsPath := path.Join("test_data", "training_test")
+
+	trainingSession, errorSessionCreation := NewTrainingSession(path.Join(artifactsPath, "checkpoint"),
+		path.Join(artifactsPath, "training_model.onnx"),
+		path.Join(artifactsPath, "eval_model.onnx"),
+		path.Join(artifactsPath, "optimizer_model.onnx"),
+		nil,
+		nil,
+		options)
+
+	defer func(session *TrainingSession) {
```

Comment:
This `defer` to destroy the session needs to be moved after the `if errorSessionCreation != nil` check.  This was what was causing my panic earlier when the `test_data` was missing---the session wasn't created properly (and an error was returned), but the test still panic'd when Destroy was called on the nil session.